core/trino-main/src/main/java/io/trino/server/remotetask/DynamicFiltersFetcher.java

```diff
@@ -113,17 +118,38 @@ public synchronized void start()
 
     public synchronized void updateDynamicFiltersVersionAndFetchIfNecessary(long newDynamicFiltersVersion)
     {
+        if (noMoreUpdateRequests) {
```

Comment:
again, we need that last update to happen to fetch final DFs.

Just make dynamic filters fetcher report when it's finished.

Reply:
see my previous comment. 

What do you exactly mean by `updateDynamicFiltersVersionAndFetchIfNecessary`. 
LIke return future from `updateDynamicFiltersVersionAndFetchIfNecessary` and mark it completed when relevant processing is completed?

Reply:
I mean essentially that the fetchers are not considered closed if there is still a pending request. It may happens that last `ContinuousTaskStatusFetcher.java` request could trigger DF fetch request, so the end condition is:
1. ContinuousTaskStatusFetcher is REALLY closed (no pending request)
2. DF doesn't have any pending request.
I think if you just put checkpoints in right places accounting for the above you should be fine